stablehlo/dialect/StablehloOps.td

```diff
@@ -356,8 +356,8 @@ def StableHLO_FloorOp: StableHLO_UnaryElementwiseOp<"floor",
 
 def StableHLO_ImagOp: StableHLO_UnaryElementwiseOp<"imag",
     [Pure, DeclareOpInterfaceMethods<InferTypeOpInterface>],
-    HLO_FpOrComplexTensor, HLO_FpTensor> {
-  let summary = "Imag operator";
+    HLO_FpOrComplexTensor /*imag_i1*/, HLO_FpTensor> {/*imag_c1*/
```

Comment:
This means we are taking in floating-point or complex as inputs, and only outputs floating-point. This is where we provide these. By default, if we provide only one, `OperandType` and `ResultType` will have the same type. Since we are extracting floating-point value (no extracting if operand is already a floating-point) from a complex element type, we would need to provide the two types separately:
```
class StableHLO_UnaryElementwiseOp<string mnemonic, list<Trait> traits,
    Type OperandType, Type ResultType = OperandType> : StableHLO_Op<mnemonic, traits # [Elementwise,
    InferShapedTypeOpInterface, SameOperandsAndResultShape]>
```

Reply:
Thanks for the explaination!